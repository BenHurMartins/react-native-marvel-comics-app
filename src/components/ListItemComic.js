import React from 'react';
import {TouchableOpacity, Image, Text, StyleSheet, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {Colors, Dimensions} from '../styles';

const ListItem = ({item, callback}) => {
  const navigation = useNavigation();
  return (
    <>
      <TouchableOpacity
        style={styles.listItemContainer}
        // onPress={() => callback(item)}
      >
        <Image
          style={styles.image}
          resizeMode={'cover'}
          height={40}
          width={40}
          source={{
            uri: `${item.thumbnail?.path.toString()}/landscape_small.jpg`.replace(
              'http://',
              'https://',
            ),
          }}
        />
        <View>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.description}>
            {`Print Price: ${item.prices[0]?.price}`}
          </Text>
        </View>
      </TouchableOpacity>
      <View style={styles.divider} />
    </>
  );
};

const styles = StyleSheet.create({
  listItemContainer: {
    // backgroundColor: Colors.PRIMARY,
    height: 50,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    flexDirection: 'row',
  },
  title: {
    color: Colors.WHITE,
    fontSize: 20,
  },
  divider: {
    width: '90%',
    height: 0.5,
    justifyContent: 'center',
    alignSelf: 'center',
    backgroundColor: Colors.PRIMARY,
  },
  image: {
    borderRadius: 20,
    borderWidth: 1,
    borderColor: Colors.PRIMARY,
    marginHorizontal: 10,
  },
  title: {
    fontSize: 15,
    fontWeight: 'bold',
    height: 20,
    width: Dimensions.SCREEN_WIDTH - 80,
  },
  description: {
    fontSize: 10,
    fontWeight: '300',
    // width: Dimensions.SCREEN_WIDTH / 2,
    height: 12,
  },
});

export default ListItem;
