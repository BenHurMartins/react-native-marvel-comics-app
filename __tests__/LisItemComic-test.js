/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render, fireEvent, act} from '@testing-library/react-native';
import ListItemComic from '../src/components/ListItemComic';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const item = {
  thumbnail: {path: 'https'},
  title: 'mock comics',
  prices: [{price: 1.99}],
};

describe('Redenring tests for ListComics Items', () => {
  it('ListComics renders correctly', () => {
    renderer.create(<ListItemComic item={item} />);
  });
  it('ListComics snapshot test', () => {
    const tree = renderer.create(<ListItemComic item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should return 1 comic in the list', () => {
    const {getByText} = render(<ListItemComic item={item} />);
    act(() => {
      expect(getByText('mock comics')).not.toBe(null);
    });
    // expect(1).to.be.equal(1);
  });
});
