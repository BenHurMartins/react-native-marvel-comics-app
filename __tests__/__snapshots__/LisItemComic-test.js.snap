// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redenring tests for ListComics Items ListComics snapshot test 1`] = `
Array [
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-start",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Image
      height={40}
      resizeMode="cover"
      source={
        Object {
          "uri": "https/landscape_small.jpg",
        }
      }
      style={
        Object {
          "borderColor": "#000000",
          "borderRadius": 20,
          "borderWidth": 1,
          "marginHorizontal": 10,
        }
      }
      width={40}
    />
    <View>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "height": 20,
            "width": 670,
          }
        }
      >
        mock comics
      </Text>
      <Text
        style={
          Object {
            "fontSize": 10,
            "fontWeight": "300",
            "height": 12,
          }
        }
      >
        Print Price: 1.99
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#000000",
        "height": 0.5,
        "justifyContent": "center",
        "width": "90%",
      }
    }
  />,
]
`;
